core/src/main/java/dk/alexandra/fresco/lib/real/AdvancedRealNumeric.java

```diff
@@ -0,0 +1,71 @@
+package dk.alexandra.fresco.lib.real;
+
+import dk.alexandra.fresco.framework.DRes;
+import dk.alexandra.fresco.framework.builder.ComputationDirectory;
+import java.math.BigDecimal;
+import java.util.List;
+
+public interface AdvancedRealNumeric extends ComputationDirectory {
+
+  /**
+   * Calculate the sum of all terms in a list.
+   * 
+   * @param terms List of secret values
+   * @return A deferred result computing the sum of the terms
+   */
+  DRes<SReal> sum(List<DRes<SReal>> terms);
+
+  /**
+   * Calculate the inner product of two secret vectors.
+   * 
+   * @param a List of secret values
+   * @param b List of secret values
+   * @return A deferred result computing computing the inner product of the two lists
+   */
+  DRes<SReal> innerProduct(List<DRes<SReal>> a, List<DRes<SReal>> b);
+
+  /**
+   * Calculate the inner product of a public and a secret vector.
+   * 
+   * @param a List of public values
+   * @param b List of secret values
+   * @return A deferred result computing computing the inner product of the two lists
+   */
+  DRes<SReal> innerProductWithPublicPart(List<BigDecimal> a, List<DRes<SReal>> b);
+
+  /**
+   * Calcualte the exponential function of a secret input x. The result is mmore precise for small
+   * inputs, so the input should be scaled if possible (note that <i>exp(n x) =
+   * exp(x)<sup>n</sup></i>).
+   * 
+   * @param x Secret value
+   * @return A deferred result computing computing e<sup>x</sup>
+   */
+  DRes<SReal> exp(DRes<SReal> x);
+
+  /**
+   * Create a random value between 0 and 1.
```

Comment:
This is still very problematic. We cannot sample such a number uniformly, so we need to document what can be expected of the sampling. Possibly, we should let the user indicate how many random bits should go into the sampling.